src/main/java/meteordevelopment/meteorclient/systems/modules/world/SpawnProofer.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * This file is part of the Meteor Client distribution (https://github.com/MeteorDevelopment/meteor-client/).
+ * Copyright (c) 2021 Meteor Development.
+ */
+
+package meteordevelopment.meteorclient.systems.modules.world;
+
+import meteordevelopment.meteorclient.events.world.TickEvent;
+import meteordevelopment.meteorclient.settings.*;
+import meteordevelopment.meteorclient.systems.modules.Categories;
+import meteordevelopment.meteorclient.systems.modules.Module;
+import meteordevelopment.meteorclient.utils.misc.Pool;
+import meteordevelopment.meteorclient.utils.player.FindItemResult;
+import meteordevelopment.meteorclient.utils.player.InvUtils;
+import meteordevelopment.meteorclient.utils.world.BlockIterator;
+import meteordevelopment.meteorclient.utils.world.BlockUtils;
+import meteordevelopment.orbit.EventHandler;
+import net.minecraft.block.*;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.registry.Registry;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SpawnProofer extends Module {
+    private final SettingGroup sgGeneral = settings.getDefaultGroup();
+
+    private final Setting<Integer> range = sgGeneral.add(new IntSetting.Builder()
+        .name("range")
+        .description("Range for block placement and rendering")
+        .min(0)
+        .sliderMax(10)
+        .defaultValue(3)
+        .build()
+    );
+
+    private final Setting<List<Block>> blocks = sgGeneral.add(new BlockListSetting.Builder()
+        .name("blocks")
+        .description("Block to use for spawn proofing")
+        .defaultValue(getDefaultBlocks())
+        .filter(this::filterBlocks)
+        .build()
+    );
+
+    private final Setting<Integer> delay = sgGeneral.add(new IntSetting.Builder()
+        .name("delay")
+        .description("Delay in ticks between placing blocks")
+        .defaultValue(0)
+        .min(0).sliderMax(10)
+        .build()
+    );
+
+    private final Setting<Boolean> rotate = sgGeneral.add(new BoolSetting.Builder()
+        .name("rotate")
+        .description("Rotates towards the blocks being placed.")
+        .defaultValue(true)
+        .build()
+    );
+
+    private final Setting<Boolean> alwaysSpawns = sgGeneral.add(new BoolSetting.Builder()
```

Comment:
would you make this an enum setting
 
Spawns {
     Always,
     Potential
}

Reply:
Actually just use the enum BlockUtils.MobSpawn